### ISSUE TITLE
Add an API similar to `GDBusMessage::from_blob`

### ISSUE THREAD

**Issue by zeenix** [dbus2/zbus#326]
In GitLab by @SeaDve on Mar 16, 2023, 15:44

Currently, there is `zbus::Message::from_raw_parts`. However, it is private. 

For the use case, [Bustle](https://github.com/freedesktop/bustle) has DBus messages serialized in a pcap file. There is currently not a convenient way to parse the bytes from each packet from the pcap file into a `zbus::Message`.

Link
* [GDBusMessage.new_from_blob](https://docs.gtk.org/gio/ctor.DBusMessage.new_from_blob.html)

**Comment by SeaDve** [comment]
I'm sorry if I took too long to get back to this. 

I am a bit puzzled about how the API should look, but my understanding of what you suggested is to have something like this or something different?

```rs
impl MessageBuilder {
    fn from_raw_parts(bytes: Vec<u8>, ...) -> Message {
        ...
    }
}
```

**Comment by zeenix** [comment]
> I am a bit puzzled about how the API should look, but my understanding of what you suggested is to have something like this, or should it be something different like returning the Builder, instead?

Out of pure coincidence I had to use something similar myself yesterday in `busd` code to create a modified copy of a message. There I used [`MessageBuilder::build_raw_body`](https://docs.rs/zbus/latest/zbus/struct.MessageBuilder.html#method.build_raw_body) combined with `TryFrom<MessageHeader> for MessageBuilder` impl I just added yesterday. However that's more building messages from separate parts instead.

For building a full Message from raw bytes. I'd just add a `TryFrom<&[u8]>` impl for `Message`.
